babel/_memoized.py

```diff
@@ -0,0 +1,36 @@
+# TODO: This can't live in .util until the circular import of
+#       core -> util -> localtime -> win32 -> core is resolved.
+
+
+class Memoized(type):
+    """
+    Metaclass for memoization based on __init__ args/kwargs.
```

Comment:
Should probably document that it assumes the constructor&initializer are pure since the cache-fetch is not locked, you could have two identical memoized instances being constructed.